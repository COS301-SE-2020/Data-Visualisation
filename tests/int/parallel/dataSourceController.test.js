/**
 * @file dataSourceController.test.js
 * Project: Data Visualisation Generator
 * Copyright: Open Source
 * Organisation: Doofenshmirtz Evil Incorporated
 * Modules: None
 * Related Documents: SRS Document - www.example.com
 * Update History:
 * Date          Author             Changes
 * -------------------------------------------------------------------------------
 * 29/06/2020   Phillip Schulze     Original
 * 30/06/2020   Phillip Schulze     Added more root modules
 *
 * Test Cases: none
 * - Testing functions that retrieve data from an Odata source
 * 		- Function that retrieves the entity list as JSON from a given Data Source
 * 		- Function that retrieves the metadata XML file for a given Data Source
 * 		- Function that retrieves entity-data as JSON from a given Data Source entity-url
 * 		- Function that tests the parsing of the XML Meta Data
 *
 * Functional Description: This file implements a snapshot, and tests if the data source controller,
 * retrieves the appropriate data.
 *
 * Error Messages: "Error"
 * Assumptions: None
 * Constraints: None
 */
/**
 * @jest-environment node
 */
const { DataSource } = require('../../../controllers/controllers');

const SRC_URL = 'https://services.odata.org/V2/Northwind/Northwind.svc';
const SRC_ENTITY = 'Products';

describe('Testing functions that retrieve data from an Odata source', () => {
	// test('Function that retrieves the entity list as JSON from a given Data Source', () => {
	// 	return DataSource.getEntityList(SRC_URL).then((list) => {
	// 		expect(list).toMatchSnapshot();
	// 	});
	// });

	// test('Function that retrieves the metadata XML file for a given Data Source', () => {
	// 	return DataSource.getMetaData(SRC_URL).then((xmlString) => {
	// 		expect(xmlString).toMatchSnapshot();
	// 	});
	// });

	// test('Function that retrieves entity-data as JSON from a given Data Source entity-url', () => {
	// 	return DataSource.getEntityData(SRC_URL, SRC_ENTITY).then((data) => {
	// 		expect(data).toMatchSnapshot();
	// 	});
	// });

	// test('Function that tests the parsing of the XML Meta Data', () => {
	// 	return DataSource.getMetaData(SRC_URL).then((meta) => {
	// 		const data = DataSource.parseMetadata(meta);
	// 		expect(data).toMatchSnapshot();
	// 	});
	// });
	test('temp', () => expect(1).toBe(1));
});
