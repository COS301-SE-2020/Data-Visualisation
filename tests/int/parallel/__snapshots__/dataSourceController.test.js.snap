// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing functions that retrieve data from an Odata source Function that retrieves entity-data as JSON from a given Data Source entity-url 1`] = `
Object {
  "data": Array [
    Array [
      1,
      undefined,
    ],
    Array [
      2,
      undefined,
    ],
    Array [
      3,
      undefined,
    ],
    Array [
      4,
      undefined,
    ],
    Array [
      5,
      undefined,
    ],
    Array [
      6,
      undefined,
    ],
    Array [
      7,
      undefined,
    ],
    Array [
      8,
      undefined,
    ],
    Array [
      9,
      undefined,
    ],
    Array [
      10,
      undefined,
    ],
    Array [
      11,
      undefined,
    ],
    Array [
      12,
      undefined,
    ],
    Array [
      13,
      undefined,
    ],
    Array [
      14,
      undefined,
    ],
    Array [
      15,
      undefined,
    ],
    Array [
      16,
      undefined,
    ],
    Array [
      17,
      undefined,
    ],
    Array [
      18,
      undefined,
    ],
    Array [
      19,
      undefined,
    ],
    Array [
      20,
      undefined,
    ],
  ],
  "entity": "Products",
  "field": undefined,
  "source": "https://services.odata.org/V2/Northwind/Northwind.svc",
}
`;

exports[`Testing functions that retrieve data from an Odata source Function that retrieves the entity list as JSON from a given Data Source 1`] = `
Object {
  "entityList": Object {
    "Alphabetical_list_of_product": Array [
      "ProductID",
      "ProductName",
      "SupplierID",
      "CategoryID",
      "QuantityPerUnit",
      "UnitPrice",
      "UnitsInStock",
      "UnitsOnOrder",
      "ReorderLevel",
      "Discontinued",
      "CategoryName",
    ],
    "Category": Array [
      "CategoryID",
      "CategoryName",
      "Description",
      "Picture",
    ],
    "Category_Sales_for_1997": Array [
      "CategoryName",
      "CategorySales",
    ],
    "Current_Product_List": Array [
      "ProductID",
      "ProductName",
    ],
    "Customer": Array [
      "CustomerID",
      "CompanyName",
      "ContactName",
      "ContactTitle",
      "Address",
      "City",
      "Region",
      "PostalCode",
      "Country",
      "Phone",
      "Fax",
    ],
    "CustomerDemographic": Array [
      "CustomerTypeID",
      "CustomerDesc",
    ],
    "Customer_and_Suppliers_by_City": Array [
      "City",
      "CompanyName",
      "ContactName",
      "Relationship",
    ],
    "Employee": Array [
      "EmployeeID",
      "LastName",
      "FirstName",
      "Title",
      "TitleOfCourtesy",
      "BirthDate",
      "HireDate",
      "Address",
      "City",
      "Region",
      "PostalCode",
      "Country",
      "HomePhone",
      "Extension",
      "Photo",
      "Notes",
      "ReportsTo",
      "PhotoPath",
    ],
    "Invoice": Array [
      "ShipName",
      "ShipAddress",
      "ShipCity",
      "ShipRegion",
      "ShipPostalCode",
      "ShipCountry",
      "CustomerID",
      "CustomerName",
      "Address",
      "City",
      "Region",
      "PostalCode",
      "Country",
      "Salesperson",
      "OrderID",
      "OrderDate",
      "RequiredDate",
      "ShippedDate",
      "ShipperName",
      "ProductID",
      "ProductName",
      "UnitPrice",
      "Quantity",
      "Discount",
      "ExtendedPrice",
      "Freight",
    ],
    "Order": Array [
      "OrderID",
      "CustomerID",
      "EmployeeID",
      "OrderDate",
      "RequiredDate",
      "ShippedDate",
      "ShipVia",
      "Freight",
      "ShipName",
      "ShipAddress",
      "ShipCity",
      "ShipRegion",
      "ShipPostalCode",
      "ShipCountry",
    ],
    "Order_Detail": Array [
      "OrderID",
      "ProductID",
      "UnitPrice",
      "Quantity",
      "Discount",
    ],
    "Order_Details_Extended": Array [
      "OrderID",
      "ProductID",
      "ProductName",
      "UnitPrice",
      "Quantity",
      "Discount",
      "ExtendedPrice",
    ],
    "Order_Subtotal": Array [
      "OrderID",
      "Subtotal",
    ],
    "Orders_Qry": Array [
      "OrderID",
      "CustomerID",
      "EmployeeID",
      "OrderDate",
      "RequiredDate",
      "ShippedDate",
      "ShipVia",
      "Freight",
      "ShipName",
      "ShipAddress",
      "ShipCity",
      "ShipRegion",
      "ShipPostalCode",
      "ShipCountry",
      "CompanyName",
      "Address",
      "City",
      "Region",
      "PostalCode",
      "Country",
    ],
    "Product": Array [
      "ProductID",
      "ProductName",
      "SupplierID",
      "CategoryID",
      "QuantityPerUnit",
      "UnitPrice",
      "UnitsInStock",
      "UnitsOnOrder",
      "ReorderLevel",
      "Discontinued",
    ],
    "Product_Sales_for_1997": Array [
      "CategoryName",
      "ProductName",
      "ProductSales",
    ],
    "Products_Above_Average_Price": Array [
      "ProductName",
      "UnitPrice",
    ],
    "Products_by_Category": Array [
      "CategoryName",
      "ProductName",
      "QuantityPerUnit",
      "UnitsInStock",
      "Discontinued",
    ],
    "Region": Array [
      "RegionID",
      "RegionDescription",
    ],
    "Sales_Totals_by_Amount": Array [
      "SaleAmount",
      "OrderID",
      "CompanyName",
      "ShippedDate",
    ],
    "Sales_by_Category": Array [
      "CategoryID",
      "CategoryName",
      "ProductName",
      "ProductSales",
    ],
    "Shipper": Array [
      "ShipperID",
      "CompanyName",
      "Phone",
    ],
    "Summary_of_Sales_by_Quarter": Array [
      "ShippedDate",
      "OrderID",
      "Subtotal",
    ],
    "Summary_of_Sales_by_Year": Array [
      "ShippedDate",
      "OrderID",
      "Subtotal",
    ],
    "Supplier": Array [
      "SupplierID",
      "CompanyName",
      "ContactName",
      "ContactTitle",
      "Address",
      "City",
      "Region",
      "PostalCode",
      "Country",
      "Phone",
      "Fax",
      "HomePage",
    ],
    "Territory": Array [
      "TerritoryID",
      "TerritoryDescription",
      "RegionID",
    ],
  },
  "source": "https://services.odata.org/V2/Northwind/Northwind.svc",
}
`;

exports[`Testing functions that retrieve data from an Odata source Function that retrieves the metadata XML file for a given Data Source 1`] = `
Object {
  "associations": Object {
    "Alphabetical_list_of_product": Array [],
    "Category": Array [
      "Products",
    ],
    "Category_Sales_for_1997": Array [],
    "Current_Product_List": Array [],
    "Customer": Array [
      "Orders",
      "CustomerDemographics",
    ],
    "CustomerDemographic": Array [
      "Customers",
    ],
    "Customer_and_Suppliers_by_City": Array [],
    "Employee": Array [
      "Employees1",
      "Employee1",
      "Orders",
      "Territories",
    ],
    "Invoice": Array [],
    "Order": Array [
      "Customer",
      "Employee",
      "Order_Details",
      "Shipper",
    ],
    "Order_Detail": Array [
      "Order",
      "Product",
    ],
    "Order_Details_Extended": Array [],
    "Order_Subtotal": Array [],
    "Orders_Qry": Array [],
    "Product": Array [
      "Category",
      "Order_Details",
      "Supplier",
    ],
    "Product_Sales_for_1997": Array [],
    "Products_Above_Average_Price": Array [],
    "Products_by_Category": Array [],
    "Region": Array [
      "Territories",
    ],
    "Sales_Totals_by_Amount": Array [],
    "Sales_by_Category": Array [],
    "Shipper": Array [
      "Orders",
    ],
    "Summary_of_Sales_by_Quarter": Array [],
    "Summary_of_Sales_by_Year": Array [],
    "Supplier": Array [
      "Products",
    ],
    "Territory": Array [
      "Region",
      "Employees",
    ],
  },
  "items": Object {
    "Alphabetical_list_of_product": Array [
      "ProductID",
      "ProductName",
      "SupplierID",
      "CategoryID",
      "QuantityPerUnit",
      "UnitPrice",
      "UnitsInStock",
      "UnitsOnOrder",
      "ReorderLevel",
      "Discontinued",
      "CategoryName",
    ],
    "Category": Array [
      "CategoryID",
      "CategoryName",
      "Description",
      "Picture",
    ],
    "Category_Sales_for_1997": Array [
      "CategoryName",
      "CategorySales",
    ],
    "Current_Product_List": Array [
      "ProductID",
      "ProductName",
    ],
    "Customer": Array [
      "CustomerID",
      "CompanyName",
      "ContactName",
      "ContactTitle",
      "Address",
      "City",
      "Region",
      "PostalCode",
      "Country",
      "Phone",
      "Fax",
    ],
    "CustomerDemographic": Array [
      "CustomerTypeID",
      "CustomerDesc",
    ],
    "Customer_and_Suppliers_by_City": Array [
      "City",
      "CompanyName",
      "ContactName",
      "Relationship",
    ],
    "Employee": Array [
      "EmployeeID",
      "LastName",
      "FirstName",
      "Title",
      "TitleOfCourtesy",
      "BirthDate",
      "HireDate",
      "Address",
      "City",
      "Region",
      "PostalCode",
      "Country",
      "HomePhone",
      "Extension",
      "Photo",
      "Notes",
      "ReportsTo",
      "PhotoPath",
    ],
    "Invoice": Array [
      "ShipName",
      "ShipAddress",
      "ShipCity",
      "ShipRegion",
      "ShipPostalCode",
      "ShipCountry",
      "CustomerID",
      "CustomerName",
      "Address",
      "City",
      "Region",
      "PostalCode",
      "Country",
      "Salesperson",
      "OrderID",
      "OrderDate",
      "RequiredDate",
      "ShippedDate",
      "ShipperName",
      "ProductID",
      "ProductName",
      "UnitPrice",
      "Quantity",
      "Discount",
      "ExtendedPrice",
      "Freight",
    ],
    "Order": Array [
      "OrderID",
      "CustomerID",
      "EmployeeID",
      "OrderDate",
      "RequiredDate",
      "ShippedDate",
      "ShipVia",
      "Freight",
      "ShipName",
      "ShipAddress",
      "ShipCity",
      "ShipRegion",
      "ShipPostalCode",
      "ShipCountry",
    ],
    "Order_Detail": Array [
      "OrderID",
      "ProductID",
      "UnitPrice",
      "Quantity",
      "Discount",
    ],
    "Order_Details_Extended": Array [
      "OrderID",
      "ProductID",
      "ProductName",
      "UnitPrice",
      "Quantity",
      "Discount",
      "ExtendedPrice",
    ],
    "Order_Subtotal": Array [
      "OrderID",
      "Subtotal",
    ],
    "Orders_Qry": Array [
      "OrderID",
      "CustomerID",
      "EmployeeID",
      "OrderDate",
      "RequiredDate",
      "ShippedDate",
      "ShipVia",
      "Freight",
      "ShipName",
      "ShipAddress",
      "ShipCity",
      "ShipRegion",
      "ShipPostalCode",
      "ShipCountry",
      "CompanyName",
      "Address",
      "City",
      "Region",
      "PostalCode",
      "Country",
    ],
    "Product": Array [
      "ProductID",
      "ProductName",
      "SupplierID",
      "CategoryID",
      "QuantityPerUnit",
      "UnitPrice",
      "UnitsInStock",
      "UnitsOnOrder",
      "ReorderLevel",
      "Discontinued",
    ],
    "Product_Sales_for_1997": Array [
      "CategoryName",
      "ProductName",
      "ProductSales",
    ],
    "Products_Above_Average_Price": Array [
      "ProductName",
      "UnitPrice",
    ],
    "Products_by_Category": Array [
      "CategoryName",
      "ProductName",
      "QuantityPerUnit",
      "UnitsInStock",
      "Discontinued",
    ],
    "Region": Array [
      "RegionID",
      "RegionDescription",
    ],
    "Sales_Totals_by_Amount": Array [
      "SaleAmount",
      "OrderID",
      "CompanyName",
      "ShippedDate",
    ],
    "Sales_by_Category": Array [
      "CategoryID",
      "CategoryName",
      "ProductName",
      "ProductSales",
    ],
    "Shipper": Array [
      "ShipperID",
      "CompanyName",
      "Phone",
    ],
    "Summary_of_Sales_by_Quarter": Array [
      "ShippedDate",
      "OrderID",
      "Subtotal",
    ],
    "Summary_of_Sales_by_Year": Array [
      "ShippedDate",
      "OrderID",
      "Subtotal",
    ],
    "Supplier": Array [
      "SupplierID",
      "CompanyName",
      "ContactName",
      "ContactTitle",
      "Address",
      "City",
      "Region",
      "PostalCode",
      "Country",
      "Phone",
      "Fax",
      "HomePage",
    ],
    "Territory": Array [
      "TerritoryID",
      "TerritoryDescription",
      "RegionID",
    ],
  },
  "sets": Array [
    "Categories",
    "CustomerDemographics",
    "Customers",
    "Employees",
    "Order_Details",
    "Orders",
    "Products",
    "Regions",
    "Shippers",
    "Suppliers",
    "Territories",
    "Alphabetical_list_of_products",
    "Category_Sales_for_1997",
    "Current_Product_Lists",
    "Customer_and_Suppliers_by_Cities",
    "Invoices",
    "Order_Details_Extendeds",
    "Order_Subtotals",
    "Orders_Qries",
    "Product_Sales_for_1997",
    "Products_Above_Average_Prices",
    "Products_by_Categories",
    "Sales_by_Categories",
    "Sales_Totals_by_Amounts",
    "Summary_of_Sales_by_Quarters",
    "Summary_of_Sales_by_Years",
  ],
  "types": Object {
    "Alphabetical_list_of_product": Array [
      "Edm.Int32",
      "Edm.String",
      "Edm.Int32",
      "Edm.Int32",
      "Edm.String",
      "Edm.Decimal",
      "Edm.Int16",
      "Edm.Int16",
      "Edm.Int16",
      "Edm.Boolean",
      "Edm.String",
    ],
    "Category": Array [
      "Edm.Int32",
      "Edm.String",
      "Edm.String",
      "Edm.Binary",
    ],
    "Category_Sales_for_1997": Array [
      "Edm.String",
      "Edm.Decimal",
    ],
    "Current_Product_List": Array [
      "Edm.Int32",
      "Edm.String",
    ],
    "Customer": Array [
      "Edm.String",
      "Edm.String",
      "Edm.String",
      "Edm.String",
      "Edm.String",
      "Edm.String",
      "Edm.String",
      "Edm.String",
      "Edm.String",
      "Edm.String",
      "Edm.String",
    ],
    "CustomerDemographic": Array [
      "Edm.String",
      "Edm.String",
    ],
    "Customer_and_Suppliers_by_City": Array [
      "Edm.String",
      "Edm.String",
      "Edm.String",
      "Edm.String",
    ],
    "Employee": Array [
      "Edm.Int32",
      "Edm.String",
      "Edm.String",
      "Edm.String",
      "Edm.String",
      "Edm.DateTime",
      "Edm.DateTime",
      "Edm.String",
      "Edm.String",
      "Edm.String",
      "Edm.String",
      "Edm.String",
      "Edm.String",
      "Edm.String",
      "Edm.Binary",
      "Edm.String",
      "Edm.Int32",
      "Edm.String",
    ],
    "Invoice": Array [
      "Edm.String",
      "Edm.String",
      "Edm.String",
      "Edm.String",
      "Edm.String",
      "Edm.String",
      "Edm.String",
      "Edm.String",
      "Edm.String",
      "Edm.String",
      "Edm.String",
      "Edm.String",
      "Edm.String",
      "Edm.String",
      "Edm.Int32",
      "Edm.DateTime",
      "Edm.DateTime",
      "Edm.DateTime",
      "Edm.String",
      "Edm.Int32",
      "Edm.String",
      "Edm.Decimal",
      "Edm.Int16",
      "Edm.Single",
      "Edm.Decimal",
      "Edm.Decimal",
    ],
    "Order": Array [
      "Edm.Int32",
      "Edm.String",
      "Edm.Int32",
      "Edm.DateTime",
      "Edm.DateTime",
      "Edm.DateTime",
      "Edm.Int32",
      "Edm.Decimal",
      "Edm.String",
      "Edm.String",
      "Edm.String",
      "Edm.String",
      "Edm.String",
      "Edm.String",
    ],
    "Order_Detail": Array [
      "Edm.Int32",
      "Edm.Int32",
      "Edm.Decimal",
      "Edm.Int16",
      "Edm.Single",
    ],
    "Order_Details_Extended": Array [
      "Edm.Int32",
      "Edm.Int32",
      "Edm.String",
      "Edm.Decimal",
      "Edm.Int16",
      "Edm.Single",
      "Edm.Decimal",
    ],
    "Order_Subtotal": Array [
      "Edm.Int32",
      "Edm.Decimal",
    ],
    "Orders_Qry": Array [
      "Edm.Int32",
      "Edm.String",
      "Edm.Int32",
      "Edm.DateTime",
      "Edm.DateTime",
      "Edm.DateTime",
      "Edm.Int32",
      "Edm.Decimal",
      "Edm.String",
      "Edm.String",
      "Edm.String",
      "Edm.String",
      "Edm.String",
      "Edm.String",
      "Edm.String",
      "Edm.String",
      "Edm.String",
      "Edm.String",
      "Edm.String",
      "Edm.String",
    ],
    "Product": Array [
      "Edm.Int32",
      "Edm.String",
      "Edm.Int32",
      "Edm.Int32",
      "Edm.String",
      "Edm.Decimal",
      "Edm.Int16",
      "Edm.Int16",
      "Edm.Int16",
      "Edm.Boolean",
    ],
    "Product_Sales_for_1997": Array [
      "Edm.String",
      "Edm.String",
      "Edm.Decimal",
    ],
    "Products_Above_Average_Price": Array [
      "Edm.String",
      "Edm.Decimal",
    ],
    "Products_by_Category": Array [
      "Edm.String",
      "Edm.String",
      "Edm.String",
      "Edm.Int16",
      "Edm.Boolean",
    ],
    "Region": Array [
      "Edm.Int32",
      "Edm.String",
    ],
    "Sales_Totals_by_Amount": Array [
      "Edm.Decimal",
      "Edm.Int32",
      "Edm.String",
      "Edm.DateTime",
    ],
    "Sales_by_Category": Array [
      "Edm.Int32",
      "Edm.String",
      "Edm.String",
      "Edm.Decimal",
    ],
    "Shipper": Array [
      "Edm.Int32",
      "Edm.String",
      "Edm.String",
    ],
    "Summary_of_Sales_by_Quarter": Array [
      "Edm.DateTime",
      "Edm.Int32",
      "Edm.Decimal",
    ],
    "Summary_of_Sales_by_Year": Array [
      "Edm.DateTime",
      "Edm.Int32",
      "Edm.Decimal",
    ],
    "Supplier": Array [
      "Edm.Int32",
      "Edm.String",
      "Edm.String",
      "Edm.String",
      "Edm.String",
      "Edm.String",
      "Edm.String",
      "Edm.String",
      "Edm.String",
      "Edm.String",
      "Edm.String",
      "Edm.String",
    ],
    "Territory": Array [
      "Edm.String",
      "Edm.String",
      "Edm.Int32",
    ],
  },
}
`;
